packages/appkit/src/tests/appkit.test.ts

```diff
@@ -446,5 +447,34 @@ describe('Base', () => {
       })
       expect(result).toBe('connector-image-url')
     })
+
+    it('should switch network when requested', async () => {
+      vi.mocked(ChainController.switchActiveNetwork).mockResolvedValue(undefined)
+      vi.mocked(ChainController.addCaipNetwork).mockImplementation(() => {})
+
+      await appKit.switchNetwork(mainnet)
+
+      expect(ChainController.switchActiveNetwork).toHaveBeenCalledWith(
+        expect.objectContaining({
+          id: mainnet.id,
+          name: mainnet.name
+        })
+      )
+
+      await appKit.switchNetwork(polygon)
+
+      expect(ChainController.addCaipNetwork).toHaveBeenCalledWith(
+        expect.objectContaining({
+          id: polygon.id,
+          name: polygon.name
+        })
+      )
+      expect(ChainController.switchActiveNetwork).toHaveBeenCalledWith(
+        expect.objectContaining({
+          id: polygon.id,
+          name: polygon.name
+        })
+      )
+    })
```

Comment:
we're missing the tests for the non-included network. IMO it should throw as per prev comment but up for discussion

Reply:
Just was thinking the same, will do it

Reply:
Well actually this is (Polygon) is non-included network. So it's testing that one as well here. @tomiir 

> it should throw as per prev comment
Couldn't get that really - you mean we should throw smth in this case?